### PR TITLE
Display path table

### PR DESCRIPTION
**Pull request by gotdairyya**
There are a few things left to do:

- [x] Fix logic for selecting/deselecting a cell
- [x] Find a better placement for the table (@JackWilb would love your opinions)
- [x] Make it possible so that the user can select what values they see for each node/edge along the path

### CODE REVIEW

**Comment by JackWilb** [comment]
This doesn't seem to be working for me. Here's what I see: 
image

I can't see the rows

File: src/components/IntermediaryNodes.vue [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <script lang="ts">
 import {
-  computed, defineComponent, onMounted, watch,
+  computed, onMounted, watch,
+  defineComponent,
```
Comment (JackWilb) [inline src/components/IntermediaryNodes.vue:4]: Let's revert this change

File: src/components/IntermediaryNodes.vue [hunk 2]
```diff
@@ -87,13 +97,24 @@ export default defineComponent({
...
 
+      cell.on('click', () => {
+        if (selectedCell !== rowData[0].cellName || !showTable.value) {
+          selectedCell = rowData[0].cellName;
+          store.commit.setSelectedConnectivityPaths(rowData);
+          showTable.value = true;
+        } else if (selectedCell === rowData[0].cellName && showTable.value) {
+          showTable.value = !showTable.value;
+        }
+      });
```
Comment (JackWilb) [inline src/components/IntermediaryNodes.vue:116]: Maybe we can set a class here that shows which cell is highlighted? Just some more visual feedback so you can see which is active

File: src/components/IntermediaryNodes.vue [hunk 3]
```diff
@@ -204,7 +226,7 @@ export default defineComponent({
 <template>
   <div
     id="intNodeDiv"
-    :style="`width: ${intNodeWidth}px;`"
+    :style="`width: 700px;`"
```
Comment (JackWilb) [inline src/components/IntermediaryNodes.vue:229]: I'm curious about this change. Why 700? and if we go with 700, we need to remove all references to `intNodeWidth` in `<script>`

File: src/components/IntermediaryNodes.vue [hunk 4]
```diff
@@ -233,4 +255,10 @@ svg >>> .rowLabels {
   font-size: 12pt;
   z-index: 100;
 }
+
+svg >>> .connectivityCell:hover {
+  cursor: pointer;
+  stroke-width: 1px;
+  stroke: black;
+}
```
Comment (JackWilb) [inline src/components/IntermediaryNodes.vue:263]: We could do something similar here for the selection as I mentioned above

File: src/components/PathTable.vue [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        <v-card-title class="pt-0 mt-0">
+          <v-text-field
+            v-model="search"
+            append-icon="mdi-magnify"
+            label="Search"
+            single-line
+            hide-details
+          />
+        </v-card-title>
+      </v-card>
```
Comment (JackWilb) [inline src/components/PathTable.vue:31]: Do these need to be in v-card wrappers? Can't we just have the text-field and style it?

File: src/components/PathTable.vue [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+              <v-row>
+                <v-col
+                  v-for="(header, i) in headers"
+                  :key="i"
+                  class="pa-0"
+                >
+                  <v-select
+                    v-model="selectedHeader[i]"
+                    :items="i % 2 ? headerEdgeSelections : headerNodeSelections"
+                    label="Attribute"
```
Comment (JackWilb) [inline src/components/PathTable.vue:51]:
```suggestion
                    label="i % 2 ? 'Edge Attribute' : 'Node Attribute'"
```

File: src/components/PathTable.vue [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+        <template
+          v-slot:top
+        >
+          <v-list dense>
+            <v-list-item dense>
+              <v-row>
+                <v-col
+                  v-for="(header, i) in headers"
+                  :key="i"
+                  class="pa-0"
```
Comment (JackWilb) [inline src/components/PathTable.vue:46]: I think this needs some x padding
```suggestion
                  class="py-0"
```

File: src/components/PathTable.vue [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+        :items="tableData"
+        :search="search"
+      >
+        <template
+          v-slot:top
+        >
+          <v-list dense>
+            <v-list-item dense>
+              <v-row>
+                <v-col
```
Comment (JackWilb) [inline src/components/PathTable.vue:43]: These v-col alignments get messed up when changing data. Can we set a constant width on the table columns so they always line up?

File: src/components/PathTable.vue [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+export default {
+  name: 'PathTable',
+
+  setup() {
+    const search = ref('');
+    const pathLength = computed(() => store.state.selectedConnectivityPaths[0].vertices.length);
+
+    const headerNodeSelections = computed(() => store.getters.nodeVariableItems);
+    const headerEdgeSelections = (store.state.network ? Object.keys(store.state.edgeAttributes).filter((varName) => !isInternalField(varName)) : ['No network']);
```
Comment (JackWilb) [inline src/components/PathTable.vue:92]: Is this used elsewhere in the codebase? Maybe we should integrate it into the store?

File: src/components/PathTable.vue (hunk 6 above)
Comment (JackWilb) [reply src/components/PathTable.vue:51]: Alternatively, you could use the `headers` list and append attribute to it like `headers[i] ${Attribute}`

**Review by JackWilb — COMMENTED** [review]
I think this PR is 95% ready to go. Thanks for your hard work to get it to this state! I had a couple of code suggestions that might clean things up a bit, but I don't think there needs to be any big changes

File: src/components/IntermediaryNodes.vue (hunk 3 above)
Comment (gotdairyya) [reply src/components/IntermediaryNodes.vue:229]: that was an old thing that I forgot to switch back to

File: src/components/IntermediaryNodes.vue (hunk 1 above)
Comment (gotdairyya) [reply src/components/IntermediaryNodes.vue]: Closed with f5202f02b539ffd5cef3c921644ab4794201711a

File: src/components/IntermediaryNodes.vue (hunk 3 above)
Comment (gotdairyya) [reply src/components/IntermediaryNodes.vue]: Closed with f5202f02b539ffd5cef3c921644ab4794201711a

File: src/components/IntermediaryNodes.vue (hunk 2 above)
Comment (gotdairyya) [reply src/components/IntermediaryNodes.vue]: good suggestion! closed with bc088054db4be78c96ef84bd7eb00e3c141ccd5b

File: src/components/IntermediaryNodes.vue (hunk 4 above)
Comment (gotdairyya) [reply src/components/IntermediaryNodes.vue:272]: closed with bc088054db4be78c96ef84bd7eb00e3c141ccd5b

File: src/components/PathTable.vue (hunk 5 above)
Comment (gotdairyya) [reply src/components/PathTable.vue:31]: so I removed the card, but I styled it a little differently, let me know what you think!

File: src/components/PathTable.vue (hunk 6 above)
Comment (gotdairyya) [reply src/components/PathTable.vue]: Addressed with cd98d15584e8c87ac42ca71cc20f1ec858aedf8a

File: src/components/PathTable.vue (hunk 5 above)
Comment (gotdairyya) [reply src/components/PathTable.vue]: Addressed with cd98d15584e8c87ac42ca71cc20f1ec858aedf8a

Comment (JackWilb) [reply src/components/PathTable.vue]: I think it's okay, but it might be better with the search on the next line down. I don't think it needs to fill the horizontal space there

File: src/components/PathTable.vue (hunk 8 above)
Comment (gotdairyya) [reply src/components/PathTable.vue:50]: should be fixed with db1f89f7ad526e99e79b035423e2f430196a8b61

File: src/components/PathTable.vue (hunk 7 above)
Comment (gotdairyya) [reply src/components/PathTable.vue]: Good suggestion!

File: src/components/PathTable.vue (hunk 5 above)
Comment (gotdairyya) [reply src/components/PathTable.vue]: I don't think the search bar needs that much space though

File: src/components/PathTable.vue (hunk 9 above)
Comment (gotdairyya) [reply src/components/PathTable.vue]: Yes it was, added it to store c80e3517e73ddec5b3d8e7e4f3f7ccde54b7a97e

File: src/components/PathTable.vue (hunk 8 above)
Comment (JackWilb) [reply src/components/PathTable.vue:50]: Woo! Perfect!

File: src/components/PathTable.vue (hunk 5 above)
Comment (JackWilb) [reply src/components/PathTable.vue]: Okay

**Review by JackWilb — APPROVED** [review]
I just updated the location that the card mounts at and fixed all the type issues 🤞🏻 Hopefully that allows it to build. If you're happy with my changes (and they look good on your machine, too) go ahead and merge 🎉
